### ISSUE TITLE
onebox-compatible documentation for promoting SunPy's features and updates

### ISSUE THREAD

**Issue by WaaallEEE** [sunpy/sunpy#5362]
Hi All, 

If you are interested in promoting current, new features and your overall development work, [helionauts.org](helionauts.org) can increase your visibility. As we work toward a Github integration in HelioNauts, our forum will make it easy to direct SunPy requests toward your Github page. One way for us to do that is to make things more visually engaging and so my request here would be to have your webpages (or some of them) compatible with "Onebox". [Onebox](https://meta.discourse.org/t/rich-link-previews-with-onebox/98088) checks the site for oEmbed or Open Graph tags to give an url-preview similar to what is used by Slack. I have no idea how that can be done in Sphinx but I have asked Onebox's dev team to give me some guidelines. Here's an [ongoing conversation](https://meta.discourse.org/t/what-makes-a-site-compatible-with-onebox-previews/190949) with them.
You will gain a LOT more visibility and not just you but also other helionauts will be able to post updates and news about SunPy. (e.g. that SunPy 3 is out?... ;-) )

**Comment by nabobalis** [comment]
Thanks for the report, I will have a look at fixing this and ensure it works for our docs and website.

**Comment by nabobalis** [comment]
I am testing a fix and using opengraph.xyz, to test this. @WaaallEEE could you try https://sunpy--5366.org.readthedocs.build/en/5366/ to see if its fixed for you?

**Comment by WaaallEEE** [comment]
> I am testing a fix and using opengraph.xyz, to test this. @WaaallEEE could you try https://sunpy--5366.org.readthedocs.build/en/5366/ to see if its fixed for you?

@nabobalis It does not seem to work, still getting the error `oEmbed / OpenGraph tags could not be found:description, image`.

**Comment by nabobalis** [comment]
As far as I am aware every tag is defined and in the html file:

```html
<!-- HTML Meta Tags -->
<title>SunPy Documentation</title>
<meta name="description" content="SunPy Documentation">

<!-- Facebook Meta Tags -->
<meta property="og:description" content="SunPy Documentation">
<meta property="og:image" content="https://raw.githubusercontent.com/sunpy/sunpy-logo/master/generated/sunpy_logo_word.png" />
<meta property="og:site_name" content="SunPy Documentation" />
<meta property="og:title" content="SunPy Documentation">
<meta property="og:type" content="website">
<meta property="og:url" content="https://docs.sunpy.org/en/stable//index.html" />

<!-- Twitter Meta Tags -->
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:description" content="SunPy Documentation">
<meta name="twitter:image" content="https://raw.githubusercontent.com/sunpy/sunpy-logo/master/generated/sunpy_logo_word.png">
<meta name="twitter:title" content="SunPy Documentation">
<meta property="twitter:domain" content="https://docs.sunpy.org/en/stable/">
<meta property="twitter:url" content="https://docs.sunpy.org/en/stable//index.html">
```

How are you testing this?

**Comment by WaaallEEE** [comment]
> How are you testing this? I was testing the URL on https://www.opengraph.xyz/ and it seems to say there are no problems.
@nabobalis by simply creating a post in Helionauts, and see what onebox displays. If the linked website is not compatible with onebox, it shows the error message I showed you. 

I will report your attempt to the Discourse's onebox dev team and see what they say.

**Comment by WaaallEEE** [comment]
Here's how I test it concretely, remember for onebox preview to activate, the link must be in its own line, not inline, as illustrated here: 

![Screen Shot 2021-05-26 at 7 34 14 PM](https://user-images.githubusercontent.com/6533571/119744231-8ce33780-be59-11eb-86df-1cc89152faf9.png)

**Comment by WaaallEEE** [comment]
See new comments from the onebox dev team at: https://meta.discourse.org/t/what-makes-a-site-compatible-with-onebox-previews/190949/3

**Comment by nabobalis** [comment]
Ok, I've updated it using the extension found by dstansby:

https://sunpy--5366.org.readthedocs.build/en/5366/guide/units-coordinates.html

**Comment by nabobalis** [comment]
I have merged a PR that fixes this, it will come out in the 3.0.1 release soon(TM).

**Comment by TheTripleV** [comment]
Leaving this here as a note for people who find their way to this page while searching for a solution.
By default, Onebox reads opengraph tags from the canonical url of your webpage. For readthedocs, this points to the stable version of your documentation. When working/testing in pull requests, this creates a situation where the opengraph tags only exist on the PR build but Onebox, following the canonical url, tries to find opengraph tags in the stable build, where they don't exist.
Adding `<meta property="og:ignore_canonical" content="true" />` to the metadata is a flag to tell Onebox to opt-out of that behavior and allows for accurate Onebox previews to be generated for non-stable versions of readthedocs documentation.
